src/Infrastructure/Services/NomadService.cs

```diff
@@ -0,0 +1,160 @@
+using Deislabs.Bindle;
+using Fermyon.Nomad.Api;
+using Fermyon.Nomad.Client;
+using Fermyon.Nomad.Model;
+using Hippo.Application.Common.Interfaces;
+using Hippo.Application.Jobs;
+using Hippo.Infrastructure.Jobs;
+using Microsoft.Extensions.Configuration;
+
+namespace Hippo.Infrastructure.Services;
+
+public class NomadService : INomadService
+{
+    private readonly JobsApi _client;
+    private readonly IConfiguration _configuration;
+
+    public NomadService(IConfiguration configuration)
+    {
+        _configuration = configuration;
+        var nomadUrl = configuration.GetValue("Nomad:Url", "http://localhost:4646/v1");
+        var nomadSecret = configuration.GetValue("Nomad:Secret", "");
+
+        Configuration config = new Configuration();
+        config.BasePath = nomadUrl;
+        config.ApiKey.Add("X-Nomad-Token", nomadSecret);
+
+        _client = new JobsApi(config);
+    }
+
+    public void StartJob(Guid id, string bindleId, Dictionary<string, string> environmentVariables, string? domain)
+    {
+        var job = new NomadJob(_configuration, id, bindleId, domain!);
+
+        foreach (var e in environmentVariables)
+        {
+            job.AddEnvironmentVariable(e.Key, e.Value);
+        }
+
+        if (DoesJobExist(id.ToString()))
+        {
+            DeleteJob(job.Id.ToString());
+        }
+        
+        PostJob(job);
+    }
+
+    public void DeleteJob(string jobName)
+    {
+        _client.DeleteJob(jobName);
+    }
+
+    private void PostJob(Application.Jobs.Job job)
+    {
+        var nomadJob = job as NomadJob;
+        var jobRegisterRequest = GenerateJobRegisterRequest(nomadJob);
```

Comment:
Job may not be a NomadJob. We should probably do a null check after line 54.

Maybe something like this (pseudo-code, may not compile):

```
_ = nomadJob ?? throw new UnsupportedException("job must be of type NomadJob");
```